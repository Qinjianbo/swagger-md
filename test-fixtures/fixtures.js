import { create as createFixtureLoader } from 'fixture-loader';

const fixture_loader = createFixtureLoader('./test-fixtures/fixtures');

const api = {

  loadSwaggerSpec(filename) {
    return fixture_loader.loadParsedJson('swagger', filename);
  },

};

export default api;
