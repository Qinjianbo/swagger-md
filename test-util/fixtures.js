import { create as createFixtureLoader } from 'fixture-loader';

const fixture_loader = createFixtureLoader('./test-util/fixtures');

const api = {

  loadSwaggerSpec(filename) {
    return fixture_loader.loadParsedJson('swagger', filename);
  },

};

export default api;
