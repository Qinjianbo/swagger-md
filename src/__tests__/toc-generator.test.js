import fixtures from '../../test-fixtures/fixtures';
import generator from '../generators/toc-generator';

describe('generateTableOfContents', () => {

  it('should return the TOC section', () => {
    const paths_header = 'Endpoints';
    const { paths } = fixtures.loadSwaggerSpec('pet-store');
    const toc = generator.generateTableOfContents(paths_header, paths);
    expect(toc).toMatchSnapshot();
  });

});
