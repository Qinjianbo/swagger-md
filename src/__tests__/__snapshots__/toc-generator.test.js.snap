exports[`generateTableOfContents should return the TOC section 1`] = `
"- [Endpoints](#endpoints)
  - [GET /pets](#get-pets)
  - [POST /pets](#post-pets)
  - [GET /pets/{id}](#get-petsid)
  - [DELETE /pets/{id}](#delete-petsid) _\`deprecated\`_"
`;
